### PR TITLE
[FRONT-132] Sort Table

### PR DESCRIPTION
**Pull request by ArnoClaude**
## Description of Issue
Linear tasks:
https://linear.app/la-famiglia-project/issue/FRONT-132/sort-functionality-for-table

## How I implemented
The sort function is more convoluted than one might think, as I cannot just grab the column's values for some columns, but have to re-compute them manually.
## Other
You'll notice that clicking on a column name in the Sort dropdown causes the dropdown to close. Now, either we keep it like that, or we disable the auto-closing, but then again, it won't close when clicking outside neither. Or I can probably make it work such that clicking outside works, but clicking inside doesn't close it, so that you see the newly applied Sort. This could get hacky and messy code though I think.

### CODE REVIEW

File: src/components/page/overview/Table.tsx [hunk 1]
```diff
@@ -53,10 +55,76 @@ const matchesFilter = (cell: Cell<Project, unknown>, filter: TableFilter) => {
...
+
+  const value1 = r1.getValue(sort.column)
+  const value2 = r2.getValue(sort.column)
+
+  if (typeof value1 === 'number' && typeof value2 === 'number') {
+    return sort.direction === 'desc' ? value2 - value1 : value1 - value2
+  }
+
+  return 0
+}
```
[the remark this answers is not in the document]
Comment (kigadye) [reply src/components/page/overview/Table.tsx:122]: also move filter function above to separate file, forgot to mention in previous review

File: src/components/page/overview/TableSortModal.tsx
Comment (kigadye) [inline src/components/page/overview/TableSortModal.tsx:1]: I think the UX would be much better if the current sorting also gets added into the filterbar (always as first entry) and this model is opened from there, not from the button at the top

**Review by kigadye — CHANGES_REQUESTED** [review]
very nice but a few improvements are possible

**Review by kigadye — APPROVED** [review]

**Review by pczern — APPROVED** [review]
we are on time limit

**Pull request merged**